### PR TITLE
Increase the timeout to run `tests/test_server.py::assert_trained_model` test successfully on Windows

### LINKED ISSUE #8658: Flakey CI test: Error empty file
**Error Information**:
```
WARNING  sanic.root:framework.py:63 Unexpected ASGI request. Forcing SPF into running mode without a server.
ERROR    rasa.model:model.py:235 Failed to extract model at C:\Users\runneradmin\AppData\Local\Temp\pytest-of-runneradmin\pytest-0\popen-gw0\test_train_with_retrieval_even0\model.tar.gz. Error: empty file
```

**Job run URL examples**:
https://github.com/RasaHQ/rasa/runs/2436006574
https://github.com/RasaHQ/rasa/runs/2500868117

**Definition of Done**:
- [ ] apply fix so that tests run deterministic

### PR DESCRIPTION
**Pull request by alwx**
**Proposed changes**:
- fix #8658 

The reason for this change: it sometimes takes longer to run the training process on Windows (everything is a bit slower on Windows GH runners), and that's why the test for `tests/test_server.py::assert_trained_model` is sometimes flakey. This change fixes it.

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: tests/test_server.py [hunk 1]
```diff
@@ -537,7 +538,7 @@ async def test_train_core_success_with(
     assert os.path.exists(os.path.join(model_path, "fingerprint.json"))
 
 
-async def test_train_with_retrieval_events_success(
```
Comment (ancalita) [inline tests/test_server.py:540]: Why did you edit the original test name? The name is now a duplicate of the test below that gets called in this test body.

**Review by ancalita — COMMENTED** [review]
Great investigation work and resolution 💯

File: tests/test_server.py (hunk 1 above)
Comment (alwx) [reply tests/test_server.py]: Oops, this is a mistake. Fixed it.

File: changelog/8658.misc.md [hunk 2]
```diff
@@ -0,0 +1 @@
+Increase the timeout to run `tests/test_server.py::assert_trained_model` test successfully on Windows.
```
Comment (ancalita) [inline changelog/8658.misc.md:1]: This could be nitpicking now - shouldn't the name of the test in the changelog also change? 😄

Comment (alwx) [reply changelog/8658.misc.md:1]: This is so stupid that the test name has somehow changed 🙈 Probably my mind was occupied with something else at this point.

Comment (alwx) [reply changelog/8658.misc.md]: Fixed now.

Comment (ancalita) [reply changelog/8658.misc.md]: No worries :)

**Review by ancalita — APPROVED** [review]
💯

**Pull request merged**